### PR TITLE
Restore user creation on Mac

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

Restore the creation of users on Mac. Resolving #521

##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

File: src/action/common/place_nix_configuration.rs [hunk 1]
```diff
@@ -57,6 +57,9 @@ impl PlaceNixConfiguration {
             "extra-nix-path".to_string(),
             "nixpkgs=flake:nixpkgs".to_string(),
         );
+
+        // Auto-allocate uids is broken on Mac. Tools like `whoami` don't work.
```
Comment (cole-h) [inline src/action/common/place_nix_configuration.rs:61]: May be a nice idea to link to the issue I filed about this:
```suggestion
        // Auto-allocate uids is broken on Mac. Tools like `whoami` don't work.
        // e.g. https://github.com/NixOS/nix/issues/8444
```

File: src/planner/macos.rs [hunk 2]
```diff
@@ -143,6 +143,11 @@ impl Planner for Macos {
                 .await
                 .map_err(PlannerError::Action)?
                 .boxed(),
+            // Auto-allocate uids is broken on Mac. Tools like `whoami` don't work.
```
Comment (cole-h) [inline src/planner/macos.rs:146]: Maybe also here

File: src/settings.rs [hunk 3]
```diff
@@ -94,6 +94,55 @@ pub struct CommonSettings {
...
+    )]
+    #[cfg_attr(
+        all(target_os = "macos", feature = "cli"),
+        clap(default_value = "_nixbld")
+    )]
+    #[cfg_attr(
+        all(target_os = "linux", feature = "cli"),
+        clap(default_value = "nixbld")
+    )]
+    pub nix_build_user_prefix: String,
```
Comment (cole-h) [inline src/settings.rs:110]: I would expect this to not be necessary for Linux with auto uids enabled... Am I wrong?

By that I mean I would expect this to be `None` on Linux and `Some(...)` on Darwin (maybe I misunderstand how we do auto uids though).

Same for all the other settings below.

Comment (Hoverbear) [reply src/settings.rs:110]: We do set the users to 0, I could move these settings to Mac only settings, and have these actions take that... But I feel like it's entirely possible we'll remove the feature on Linux still so I don't want to possibly make a bunch of work for ourselves later.

**Review by cole-h — APPROVED** [review]
:sob: :+1:

**Comment by Hoverbear** [comment]
I know! It's so sad. :(

**Pull request merged**